### PR TITLE
OIA-55: Add config extension for collectors

### PR DESCRIPTION
**Pull request by fooker**

### CODE REVIEW

File: config/src/main/java/org/opennms/integration/api/xml/schema/collector/CollectorXml.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    @XmlElement(name="parameter")
+    private List<ParameterXml> parameters = new ArrayList<>();
+
+    public String getService() {
+        return this.service;
+    }
+
+    public void setService(final String service) {
+        this.service = service;
```
Comment (amay94) [inline config/src/main/java/org/opennms/integration/api/xml/schema/collector/CollectorXml.java:63]: Should we require non-null for service and class name here?

File: config/src/main/java/org/opennms/integration/api/xml/schema/collector/ParameterXml.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    @XmlAttribute(name="value")
+    private String value;
+
+    public String getKey() {
+        return this.key;
+    }
+
+    public void setKey(final String key) {
+        this.key = key;
```
Comment (amay94) [inline config/src/main/java/org/opennms/integration/api/xml/schema/collector/ParameterXml.java:55]: Same question about null checks here

**Review by RangerRick — APPROVED** [review]
obviously it's hard to be definitive about approving without an impl with tests, but this looks pretty good to me 👍

**Review by amay94 — APPROVED** [review]
Looks good, only minor questions

File: config/src/main/java/org/opennms/integration/api/xml/schema/collector/CollectorXml.java (hunk 1 above)
Comment (fooker) [reply config/src/main/java/org/opennms/integration/api/xml/schema/collector/CollectorXml.java:63]: I would consider this a pure data class. As the constructor is not enforcing a value in the field, I would not do so in the setter.

**Pull request merged**